spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -139,40 +133,6 @@ public SimpleJobService(SearchableJobInstanceDao jobInstanceDao, SearchableJobEx
 
 		stepExecutionDao.addStepExecutions(jobExecution);
 
-		String jobName = jobExecution.getJobInstance() == null ? jobInstanceDao.getJobInstance(jobExecution).getJobName() : jobExecution.getJobInstance().getJobName();
-		Collection<String> missingStepNames = new LinkedHashSet<String>();
-
-		if (jobName != null) {
```

Comment:
It would be good if we know why we added this logic of finding missing step names at first place.

Reply:
@ilayaperumalg Great Question! This looks to be a copy paste from Spring Batch Admin https://github.com/spring-attic/spring-batch-admin/blob/master/spring-batch-admin-manager/src/main/java/org/springframework/batch/admin/service/SimpleJobService.java.  This was pulled in as a part of SCDF 2.2.0 when we removed the need to have the Spring Batch Admin. 
SCDF never used these features and thus only passed in the `MapJobRegistry` just to fulfill the requirement of the `SimpleJobServiceFactoryBean` to have a `JobLocator`.   This can be seen in the SCDF 1.7 release.

Reply:
@cppwfs Thanks for the description!